Mesh4all/m4a-firmware
# wifi-subsys: Add Component Protocols -> Group of network protocols

### Contribution description
This Pr sort all the network's protocols inside of the network component and move it to a new component called protocols, this will be save sntp, icmp, and https protocols. 
  



### Testing procedure

run this command inside of wifi-subsys/tests folder.

```sh
 idf.py build -D "TEST_COMPONENTS=protocols" flash monitor
```

Recommends before to run the tests.

- Comment the main.c 

```c
void app_main(void) {

    // unity_run_all_tests();  --> This function should be commented in this case to decide any test
    // //
    printf("Starting interactive test menu\n");
    unity_run_menu();
}
```

- Go to the main folder of wifi-subsys  (wifi-subsys/main) and check if you has an dev-config.h header file, is you did'nt has one,  copy the default_params.h and changes all required test params as WSTA_SSID and WSTA_PASS, ( To icmp you need to include DEFAULT_SERVER_TO_PING and define it ).

- Remember change USE_DEV_CONFIG to 1 in default_params.h to load all your settings changed in the last step.

Expected Output: 

- Press enter 
```sh
Here's the test menu, pick your combo:
(1)	"Wi-Fi initialize" [network]
(2)	"setting ping session" [network]
(3)	"send manual ping" [network]
(4)	"get current status" [network]
(5)	"Get and set time by sntp" [network][protocols][sntp]
```



### Issues/PRs references
Fixes #140
